beaker_kernel/lib/context.py

```diff
@@ -41,12 +41,12 @@ def __init__(self, beaker_kernel: "LLMKernel", agent_cls: "BaseAgent", config: D
         self.jinja_env = None
         self.templates = {}
         self.beaker_kernel = beaker_kernel
+        self.config = config
+        self.subkernel = self.get_subkernel()
         self.agent = agent_cls(
             context=self,
-            tools=[],
+            tools=[self.subkernel],
```

Comment:
What happens to the prompting if you add a class that doesn't have any tools?

Reply:
https://github.com/jataware/archytas/blob/36c8ace7400b0b81abb24e7fa77a75289b6b0c5e/archytas/tool_utils.py#L41-L50

I'll double check, but if an empty class somehow breaks an issue needs to be opened up in archytas.

Reply:
I think it is pretty unlikely to break things, as in to cause errors, but I think that it could change the prompt that the archytas react agent generates.
Look in to how the prompt is generated around tools.

Reply:
When pulling the info for the comment I just left, I saw that there isn't a problem if there are no tools defined. 
i.e.:
```
        tool_methods = collect_tools_from_object(obj)
        if tool_methods:
          # All the stuff that changes the prompt
```

Reply:
I'm going to resolve this thread because I think it's very similar to https://github.com/jataware/beaker-kernel/pull/47#discussion_r1594587681

Feel free to reopen though